File: Classification/Logistic Regression/Advertisement.R
#Import dataset
source = read.csv('Social_Network_ads.csv')
source = source[3:5]

#Split dataset
#library(caTools)
split = sample.split(source$Purchased, SplitRatio = 0.25)
training_set = subset(source, split == FALSE)
test_set = subset(source, split == TRUE)

#Feature scaling
training_set[1:2] = scale(training_set[1:2])
test_set[-3] = scale(test_set[-3])

#Creating a model
model = glm(formula = Purchased ~ .,
            family= binomial,
            data = training_set)
#Prediction
prediction_probability = predict(model, type='response', newdata = test_set[1:2])
y_pred = ifelse(prediction_probability > 0.5, 1,0)

#Confusion matrix
cm = table(test_set[,3],y_pred > 0.5)

# Visualising the Training set results
install.packages('ElemStatLearn')
library(ElemStatLearn)
set = training_set
x1 = seq(min(set[, 1]) - 1, max(set[, 1]) + 1, by = 0.01)
x2 = seq(min(set[, 2]) - 1, max(set[, 2]) + 1, by = 0.01)
grid_set = expand.grid(x1, x2)
colnames(grid_set) = c('Age', 'EstimatedSalary')
prob_set = predict(model, type = 'response', newdata = grid_set)
y_grid = ifelse(prob_set > 0.5, 1, 0)
plot(set[, -3],
     main = 'Logistic Regression (Training set)',
     xlab = 'Age', ylab = 'Estimated Salary',
     xlim = range(x1), ylim = range(x2))
contour(x1, x2, matrix(as.numeric(y_grid), length(x1), length(x2)), add = TRUE)
points(grid_set, pch = '.', col = ifelse(y_grid == 1, 'springgreen3', 'tomato'))
points(set, pch = 21, bg = ifelse(set[, 3] == 1, 'green4', 'red3'))

# Visualising the Test set results
library(ElemStatLearn)
set = test_set
x1 = seq(min(set[, 1]) - 1, max(set[, 1]) + 1, by = 0.01)
x2 = seq(min(set[, 2]) - 1, max(set[, 2]) + 1, by = 0.01)
grid_set = expand.grid(X1, X2)
colnames(grid_set) = c('Age', 'Salary')
prob_set = predict(classifier, type = 'response', newdata = grid_set)
y_grid = ifelse(prob_set > 0.5, 1, 0)
plot(set[, -3],
     main = 'Logistic Regression (Test set)',
     xlab = 'Age', ylab = 'Estimated Salary',
     xlim = range(x1), ylim = range(x2))
contour(x1, x2, matrix(as.numeric(y_grid), length(X1), length(X2)), add = TRUE)
points(grid_set, pch = '.', col = ifelse(y_grid == 1, 'springgreen3', 'tomato'))
points(set, pch = 21, bg = ifelse(set[, 3] == 1, 'green4', 'red3'))

File: Classification/Logistic Regression/Advertisement.py
#Import libraries
import pandas as p
import numpy as n
import matplotlib.pyplot as visuals

#Import libraries
source = p.read_csv('Social_Network_Ads.csv')
x= source.iloc[:,2:4].values
y= source.iloc[:,4].values

#Splitting the data
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.25)

#Feature Scaling
from sklearn.preprocessing import StandardScaler
fScale = StandardScaler()
x_train = fScale.fit_transform(x_train)
x_test = fScale.transform(x_test)

#Creating a model
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model.fit(x_train,y_train)

#Prediction
y_pred = model.predict(x_test)

#Confusion matrix
from sklearn.metrics import confusion_matrix
cm= confusion_matrix(y_test,y_pred)

#Visualization
from matplotlib.colors import ListedColormap
x_set,y_set = x_train, y_train
x1,x2 = n.meshgrid(n.arange(start = x_set[:,0].min() -1, stop = x_set[:,0].max()+1, step = 0.001),
                    n.arange(start = x_set[:,1].min() -1, stop = x_set[:,1].max()+1, step = 0.001))
visuals.contourf(x1,x2, model.predict(n.array([x1.ravel(), x2.ravel()]).T).reshape(x1.shape),
                 alpha = 0.75, cmap = ListedColormap(('red','green')))
visuals.xlim(x1.min(), x1.max())
visuals.ylim(x2.min(), x2.max())
for i,j in enumerate(n.unique(y_set)):
    visuals.scatter(x_set[y_set ==j, 0], x_set[y_set ==j,1],
    c= ListedColormap(('red','green'))(i), label = j)
visuals.title('Predict uers who buy certain item - Logistic Regression')
visuals.xlabel('Age')
visuals.ylabel('Salary')
visuals.legend()
visuals.show()

# Visualising the Test set results
from matplotlib.colors import ListedColormap
x_set, y_set = x_test, y_test
x1, x2 = n.meshgrid(n.arange(start = x_set[:, 0].min() - 1, stop = x_set[:, 0].max() + 1, step = 0.01),
                     n.arange(start = x_set[:, 1].min() - 1, stop = x_set[:, 1].max() + 1, step = 0.01))
visuals.contourf(x1, x2, model.predict(n.array([x1.ravel(), x2.ravel()]).T).reshape(x1.shape),
             alpha = 0.75, cmap = ListedColormap(('red', 'green')))
visuals.xlim(x1.min(), x1.max())
visuals.ylim(x2.min(), x2.max())
for i, j in enumerate(n.unique(y_set)):
    visuals.scatter(x_set[y_set == j, 0], x_set[y_set == j, 1],
                c = ListedColormap(('red', 'green'))(i), label = j)
visuals.title('Predict uers who buy certain item - Logistic Regression')
visuals.xlabel('Age')
visuals.ylabel('Salary')
visuals.legend()
visuals.show()